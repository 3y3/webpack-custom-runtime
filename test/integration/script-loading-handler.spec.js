'use strict';

const cct = require('./utils/custom-configurable-test');
const WebpackCustomRuntime = require('../../').CustomRuntimePlugin;

class TestPlainLoadingHandler extends WebpackCustomRuntime.LoadingHandler {

    codeResolver(options, chunk, hash, { result }) {
        return `console.log(${result});`;
    }
}

class TestFuncLoadingHandler extends WebpackCustomRuntime.LoadingHandler {

    codeResolver() {
        return function(p1, p2, p3) {
            console.log(p3);
        }
    }
}

describe('scriptLoadingHandler', function() {

    function cit(name, plugins = []) {
        cct(name, {
            plugins: [
                new WebpackCustomRuntime({ behavior: '' }),
                ...plugins
            ]
        });
    }

    cit('should apply custom plain loading handler', [
        new TestPlainLoadingHandler()
    ]);

    cit('should apply custom func loading handler', [
        new TestFuncLoadingHandler()
    ]);

    cit('should apply multy loading handlers', [
        new TestPlainLoadingHandler(),
        new TestFuncLoadingHandler()
    ]);
});
