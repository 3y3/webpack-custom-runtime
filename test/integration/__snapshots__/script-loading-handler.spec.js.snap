// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scriptLoadingHandler should apply custom func loading handler: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ Change 1 @@
-/******/ 	  return Promise.resolve();
+/******/ 	  return [
+/******/ 	    function(p1, p2, p3) {
+/******/ 	      console.log(p3);
+/******/ 	    }
+/******/ 	  ].reduce(function(promise, strategy) {
+/******/ 	    return promise.then(function(result) {
+/******/ 	      return strategy(installedChunks, chunkId, result);
+/******/ 	    });
+/******/ 	  }, Promise.resolve(result));
"
`;

exports[`scriptLoadingHandler should apply custom plain loading handler: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ Change 1 @@
-/******/ 	  return Promise.resolve();
+/******/ 	  return [
+/******/ 	    function(installedChunks, chunkId, result) {
+/******/ 	      console.log(result);
+/******/ 	    }
+/******/ 	  ].reduce(function(promise, strategy) {
+/******/ 	    return promise.then(function(result) {
+/******/ 	      return strategy(installedChunks, chunkId, result);
+/******/ 	    });
+/******/ 	  }, Promise.resolve(result));
"
`;

exports[`scriptLoadingHandler should apply multy loading handlers: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ Change 1 @@
-/******/ 	  return Promise.resolve();
+/******/ 	  return [
+/******/ 	    function(installedChunks, chunkId, result) {
+/******/ 	      console.log(result);
+/******/ 	    },
+/******/ 	    function(p1, p2, p3) {
+/******/ 	      console.log(p3);
+/******/ 	    }
+/******/ 	  ].reduce(function(promise, strategy) {
+/******/ 	    return promise.then(function(result) {
+/******/ 	      return strategy(installedChunks, chunkId, result);
+/******/ 	    });
+/******/ 	  }, Promise.resolve(result));
"
`;
